package/MDAnalysis/analysis/base.py

```diff
@@ -389,8 +421,8 @@ def _conclude(self):
         """
         pass  # pylint: disable=unnecessary-pass
 
-    def run(self, start=None, stop=None, step=None, frames=None,
-            verbose=None, *, progressbar_kwargs={}):
+    def _compute(self, start=None, stop=None, step=None, 
```

Comment:
Maybe it's not necessary to have all args parsed from `run` to compute if you run `_setup_frames` in the current `run` instead.

Reply:
I don't understand what you're proposing :(

Reply:
I mean you can
```python
def _compute(self, verbose=None, *, progressbar_kwargs={}):
   ...

def run(self, start=None, stop=None, step=None, frames=None,
            verbose=None, scheduler=None, 
            *, progressbar_kwargs={},
            ):

        self._setup_frames(self._trajectory, start=start, stop=stop,
                           step=step, frames=frames)
    ...
```
because `_setup_frames` will store these values as attributes.

Besides, you need to make sure `self.start` etc are assigned to what the user has requested (not just the last bslice)

Reply:
done, I guess? `_setup_frames` is now called inside `run` instead of `_compute`